pkg/operator/staticpod/installerpod/cmd.go

```diff
@@ -300,55 +300,37 @@ func (o *InstallOptions) copyContent(ctx context.Context) error {
 		if _, exists := podConfigMap.Data["pod.yaml"]; !exists {
 			return true, fmt.Errorf("required 'pod.yaml' key does not exist in configmap")
 		}
-		podConfigMap = o.substituteConfigMap(podConfigMap)
-		rawPodBytes = podConfigMap.Data["pod.yaml"]
+		podsConfigMap = o.substituteConfigMap(podConfigMap)
 		return true, nil
 	})
 	if err != nil {
 		return err
 	}
-	// the kubelet has a bug that prevents graceful termination from working on static pods with the same name, filename
-	// and uuid.  By setting the pod UID we can work around the kubelet bug and get our graceful termination honored.
-	// Per the node team, this is hard to fix in the kubelet, though it will affect all static pods.
-	pod, err := resourceread.ReadPodV1([]byte(rawPodBytes))
-	if err != nil {
-		return err
-	}
-	pod.UID = uuid.NewUUID()
-	for _, fn := range o.PodMutationFns {
-		klog.V(2).Infof("Customizing static pod ...")
-		pod = pod.DeepCopy()
-		if err := fn(pod); err != nil {
-			return err
-		}
-	}
-	finalPodBytes := resourceread.WritePodV1OrDie(pod)
-
-	// Write secrets, config maps and pod to disk
-	// This does not need timeout, instead we should fail hard when we are not able to write.
-	podFileName := o.PodConfigMapNamePrefix + ".yaml"
-	klog.Infof("Writing pod manifest %q ...", path.Join(resourceDir, podFileName))
-	if err := ioutil.WriteFile(path.Join(resourceDir, podFileName), []byte(finalPodBytes), 0644); err != nil {
-		return err
-	}
 
-	// copy static pod
+	// at this point we know that the required key is present in the config map, just make sure the manifest dir actually exists
 	klog.Infof("Creating directory for static pod manifest %q ...", o.PodManifestDir)
 	if err := os.MkdirAll(o.PodManifestDir, 0755); err != nil {
 		return err
 	}
 
-	// remove the existing file to ensure kubelet gets "create" event from inotify watchers
-	if err := os.Remove(path.Join(o.PodManifestDir, podFileName)); err == nil {
-		klog.Infof("Removed existing static pod manifest %q ...", path.Join(o.PodManifestDir, podFileName))
-	} else if !os.IsNotExist(err) {
-		return err
-	}
-	klog.Infof("Writing static pod manifest %q ...\n%s", path.Join(o.PodManifestDir, podFileName), finalPodBytes)
-	if err := ioutil.WriteFile(path.Join(o.PodManifestDir, podFileName), []byte(finalPodBytes), 0644); err != nil {
-		return err
-	}
+	// then write the required pod and all optional
+	// the key must contain pod.yaml to be considered
+	for rawPodKey, rawPod := range podsConfigMap.Data {
+		if !strings.Contains(rawPodKey, "pod.yaml") {
+			continue
+		}
+		var podName string
+		if rawPodKey == "pod.yaml" {
+			// this is our main pod, assign the previous name for compatibility
+			podName = o.PodConfigMapNamePrefix + ".yaml"
+		}
```

Comment:
can't we have a fully qualified name as key? e.g. kube-apiserver-pod.yaml.

Reply:
oh, sure, we could, good idea, will change.

Reply:
I think that this is a two-step process, first, we need to merge this PR, and then we need to update `kas-o`, right ?

Reply:
probably, yes.

Reply:
ok, I will prepare a follow up PRs